packages/venia-concept/src/RootComponents/CMS/CMS.js

```diff
@@ -1,8 +1,37 @@
 import React, { Component } from 'react';
 import CategoryList from 'src/components/CategoryList';
 
+// DEBUG
+import { useEffect } from 'react';
+import { useRestApi } from '@magento/peregrine';
+import { loadingIndicator } from 'src/components/LoadingIndicator';
+
+const TestComponent = () => {
+    const [restResponse, restApi] = useRestApi('/rest/V1/guest-carts');
+    const { data, error, loading } = restResponse;
+    const { sendRequest } = restApi;
+
+    useEffect(() => {
+        const signIn = async () => {
+            await sendRequest({
+                options: {
+                    method: 'POST'
+                }
+            });
+        };
+
+        signIn();
+    }, []);
```

Comment:
Ideally we'd just 

```
useEffect(async () => {
  await sendRequest(...)
});
```

but `async` results in the anonymous function passed to `useEffect` to return a `Promise` when the function passed to `useEffect` is supposed to return an "unsubscribe" callback.

The pattern above is recommended by [Dan Abramov](https://github.com/facebook/react/issues/14326#issuecomment-441680293) and is a temporary one until the React team releases the real solution (`Suspense`).

Reply:
I had to do this for the braintree stuff, and it gets even more complicated when you have potential side effects of your `await` call that require the component to be mounted :D